xc7/fasm2bels/verilog_modeling.py

```diff
@@ -1245,12 +1248,110 @@ def __init__(self, db, grid, conn, name="top"):
         # IO bank lookup (if part was provided).
         self.iobank_lookup = {}
 
+    def set_default_iostandard(self, iostandard):
+        self.default_iostandard = iostandard
+
+    def make_iosettings_map(self, parsed_eblif):
+        """
+        Builds map of top-level port names to cell names and cell names to IO
+        # settings map using data from a parsed EBLIF file.
+        """
+
+        # Map of EBLIF cell types of IO buffers to their ports connecting them
+        # with IO pads.
+        IOBUF_PORTS = {
+            "IBUF_VPR": ["I"],
```

Comment:
There should **not** be an assumption that the input bitstream came from VPR.  This map should not be contained with `fasm2bels` at all, but a tool in front of it.  That tool can assume VPR.  That tool can then be adapted for nextpnr or Vivado itself as needed.

Reply:
The assumption is about the eblif, not the bitstream. Since Vivado cannot write eblifs its reasonable. But now when I gave some thoughts to it I have a better idea.

Reply:
Ya, the new code looks much better